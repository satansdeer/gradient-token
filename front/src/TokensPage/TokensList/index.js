import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { PropTypes as MobxPropTypes } from "mobx-react";
import { inject, observer } from "mobx-react";

import TokenItem from "./TokenItem";
import TokenView from "./TokenView";
import "./TokensList.css";

const TokensList = ({
  gradientTokenStore: { tokens },
  modalStore: { showModal }
}) => {
  return (
    <Fragment>
      {tokens.length ? (
        <div className="TokensList">
          {tokens.map(token => (
            <TokenItem
              key={token.index}
              token={token.gradient}
              onClick={() => showModal(<TokenView token={token.gradient} />)}
            />
          ))}
        </div>
      ) : (
        <div className="TokensList-label_empty">You don't have tokens yet.</div>
      )}
    </Fragment>
  );
};

TokensList.propTypes = {
  gradientTokenStore: PropTypes.shape({
    tokens: MobxPropTypes.arrayOrObservableArrayOf(PropTypes.object)
  }),
  modalStore: PropTypes.shape({
    showModal: PropTypes.func
  })
};

export default inject("gradientTokenStore", "modalStore")(observer(TokensList));
