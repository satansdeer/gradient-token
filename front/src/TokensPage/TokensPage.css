.TokensPage {
  margin-top: 50px;
}

.TokensPage-tokens {
  max-width: 800px;
  margin: 50px auto;
}
