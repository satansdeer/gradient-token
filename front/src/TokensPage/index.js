import React from "react";
import Button from "components/Button";
import PropTypes from "prop-types";
import WithLoader from "components/WithLoader";
import TokensList from "./TokensList";
import { inject, observer } from "mobx-react";
import "./TokensPage.css";

const TokensPage = ({ gradientTokenStore: { mintToken, isLoading } }) => {
  return (
    <div className="TokensPage">
      <h1>Gradient Tokens</h1>
      <Button onClick={mintToken} label="Mint token" />
      <div className="TokensPage-tokens">
        <WithLoader isLoading={isLoading}>
          <TokensList />
        </WithLoader>
      </div>
    </div>
  );
};

TokensPage.propTypes = {
  gradientTokenStore: PropTypes.shape({
    mintToken: PropTypes.func
  }).isRequired
};

export default inject("gradientTokenStore")(observer(TokensPage));
