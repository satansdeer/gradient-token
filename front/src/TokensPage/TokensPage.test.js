import React from "react";
import { shallow } from "enzyme";
import { Provider } from "mobx-react";
import renderer from "react-test-renderer";
import TokensPage from "./index";

describe("<TokensPage />", () => {
  const gradientTokenStore = {
    mintToken: jest.fn(),
    tokens: []
  };
  const stores = { gradientTokenStore };

  it("renders and matches the snapshot", () => {
    const component = renderer.create(
      <Provider {...stores}>
        <TokensPage />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
