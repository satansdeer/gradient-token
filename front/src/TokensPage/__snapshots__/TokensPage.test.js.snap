// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokensPage /> renders and matches the snapshot 1`] = `
<div
  className="TokensPage"
>
  <h1>
    Gradient Tokens
  </h1>
  <button
    className="Button undefined"
    onClick={[Function]}
  >
    Mint token
  </button>
  <div
    className="TokensPage-tokens"
  >
    <div
      className="TokensList-label_empty"
    >
      You don't have tokens yet.
    </div>
  </div>
</div>
`;
